spec/models/dynamic_field_spec.rb

```diff
@@ -0,0 +1,186 @@
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'String key can\'t be blank'
+      end
+    end
+
+    context 'when missing field_type' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, field_type: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Field type can\'t be blank'
+      end
+    end
+
+    context 'when field_type invalid' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, field_type: 'not-valid') }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Field type is not included in the list'
+      end
+    end
+
+    context 'when missing dynamic_field_group' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, dynamic_field_group: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Dynamic field group must exist'
+      end
+    end
+
+    context 'when missing created_by' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, created_by: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Created by must exist'
+      end
+    end
+
+    context 'when missing updated_by' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, updated_by: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Updated by must exist'
+      end
+    end
+
+    context 'when creating select dynamic field' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, field_type: DynamicField::Type::SELECT) }
+
+      it 'requires select_options' do
+        expect(dynamic_field.save).to be false
+        expect(dynamic_field.errors.full_messages).to include 'Select options can\'t be blank'
+      end
+    end
+
+    context 'when creating controlled term dynamic field' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, controlled_vocabulary: nil) }
+
+      it 'requires controlled_vocabulary' do
+        expect(dynamic_field.save).to be false
+        expect(dynamic_field.errors.full_messages).to include 'Controlled vocabulary can\'t be blank'
```

Comment:
```suggestion
        expect(dynamic_field.errors.full_messages).to include 'Controlled vocabulary is required'
```

Reply:
I think some of these new error messages are helpful, and I think I will implement some of them. I would like to keep the defaults in some cases tho. It think its just more code to maintain that is only beneficial in uncommon situations.

Reply:
This change doesn't bother me all that much, so I'm fine with keeping as is, and I get what you mean about the extra code.

Reply:
Cool, thanks!